Automattic/wp-calypso
# Add VS Code workspace settings

It’s always useful to have `extensions.json` and also `settings.json` inside `.vscode` directory to have a basic/common configuration to properly integrate the tools with VS Code. A good set of extensions can make working with a workspace more productive and also improve the onboarding experience.

See p1663073930100959-slack-C02DQP0FP

#### Proposed Changes

* Remove `.vscode` from `.gitignore`
* Add recommended extensions via `.vscode/extensions.json`
* Add basic workspace settings via `.vscode/settings-sample.jsonc`
* Add a node script to setup/link vs code settings
* Execute the new node script on `postinstall`

#### Testing Instructions



* Remove all your personal preferences in VS Code and make a copy of it.
    * Open `settings.json` (`Cmd + Shift + P` > `Preferences: Open Settings (JSON)`)
    * Copy the contents to a file on your computer
    * Remove all the contents of `settings.json`
    * Don't forget to restore its contents after completing your testing of this PR.
* Check out this PR and ensure to run `yarn install` and `yarn vscode:link`
* Open in VS Code or reload it (`Cmd + Shift + P` > `Developer: Reload Window`)
* Welcome to the "Actual VS Code" which you may not recognize 😄 
* Open a `.scss` file e.g. `client/components/async-load/style.scss`
* Mess up the formatting e.g. add spaces inside parentheses or use spaces instead of tabs for indentation.
* Save the file
* Confirm that the lints are auto-fixed.
* Open a JS/JSX file e.g. `client/components/action-card/index.jsx`
* Mess up the formatting
* Save the file
* Confirm that the lints are auto-fixed

#### Pre-merge Checklist




Node script results
<img width="958" alt="Screenshot 2022-09-16 at 12 18 36 PM" src="https://user-images.githubusercontent.com/18226415/190578440-6220eacc-5863-4c62-8db7-99411e1a5f16.png">
